src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -292,7 +327,14 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         for (uint8_t i = 0; i < TableColorNames.size(); ++i)
         {
-            _CurrentColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            if (i < ColorTableDivider)
+            {
+                _CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            }
+            else
+            {
+                _CurrentBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            }
         }
```

Comment:
I think you made a mistake here.

Lets say you jave 8 ColorNames, your devider is set to 4 and you have split the 8 colors in two arrays where each array has 4 colors. Then you get an IndexOutOfBounds-Exception when you try to get index 3 to 7 from the second array, because each array has only an index of 0 - 3. Imagine you have splitted 8 colours in two arrays.

I think you have to count each array on it's own. The first from 0 to `Devider - 1` and the second from 0 to `ColorNames.Count - Devider - 1`.
_ATTENTION: Don't use `Devider - 1` as max count for both arrays. If the second array has more colors because the `ColorNames.Count - Diveder` don't result in the same number of colors for both arrays, they get a different length._

Reply:
Maybe it's a better way to split the ColorNames array in two arrays too. Then you avoid mistakes on adding other colors because of moving/changing the order (especially on adding non-brigth colors).

Reply:
@htcfreek Thank you, I'll look into it.

Reply:
@htcfreek Sorry, I'm not quite sure I understand you. Would you please elaborate on what I should do to fix this? Thank you.

Reply:
> @htcfreek Sorry, I'm not quite sure I understand you. Would you please elaborate on what I should do to fix this? Thank you.

My first or my second comment?
I'll answer the details tomorrow.

Reply:
@htcfreek My first comment. Thank you.

Reply:
Have looked into it and tomorrow/later this day I will post sample code with explaination.

Reply:
> 
> 
> Have looked into it and tomorrow/later this day I will post sample code with explaination.

Thanks

Reply:
@Chips1234 
Sorry for let you waiting. But I hope I can help you, and you understand what I mean with the explanations at the end of this comment.

## My suggested code idea
⚠ Attention: I haven't run the code, and you have to check if the syntax of the mathematical operation is correct. (I am not so familiar with this.)
```cpp
        for (uint8_t i = 0; i < ColorTableDivider; ++i)
        {
            // Here, the code is taking the color from the color scheme object's table, and throwing it for non-bright colors into the "_CurrentNonBrightColorTable" list and for bright colors into the "_CurrentBrightColorTable" list. 
            // We have 16 colors, whereas 8 of them are non-bright and 8 of them are bright.
            // At the beginning of this file we set a constant variable ("ColorTableDivider") to the number of elements (colors) where we split the "TableColorNames" list.
			
            // Because of these facts we can use the value of "ColorTableDivider" as stop condition in our for loop.
            // And we can use the loop's index as index for the element in the "_CurrentNonBrightColorTable" list and the "_CurrentBrightColorTable" list we are working on at each iteration.
            // To get the correct elements from the color scheme object's table we can set/calculate the index of the non-bright color and the bright color mathematical (based on index and "ColorTableDivider") on each iteration.

            // Fill the element of "_CurrentNonBrightColorTable" with the non-brigth color (lower index of color scheme object's table: 0-7).
            int indexNonBrightColor = i;
            _CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[indexNonBrightColor]);
			
            // Fill the element of "_CurrentBrightColorTable" with the brigth color (higher index of color scheme object's table: 8-15).
            int indexBrightColor = i + ColorTableDivider;
            _CurrentBrightColorTable.GetAt(i).Color(colorScheme.Table()[indexBrightColor]);
        }
```

## Explanation
### Your code
In your code you made the mistake to iterate based on the 16 colors in "TableColorNames". This leads to an index from 0 to 15. But at the beginning of this file you separate the 16 colors into two lists with 8 colors on each list. So your condition is incorrect.
```cpp
for (uint8_t i = 0; i < TableColorNames.size(); ++i)
```

Within the loop (iteration from 0 to 15) you first tried to fill the "_CurrentNonBrightColorTable" with the first 8 colors (iteration/index 0-7) and that works because your "_CurrentNonBrightColorTable" has 8 elements (0-7).
But then you tried to fill the "_CurrentBrightColorTable" with colors based on the remaining iterations (index 8 - 15). And this CAN'T work because your "_CurrentBrightColorTable" list has only 8 elements (0-7). Remember you have split the 16 colors into two lists of 8 elements per list.
```cpp
if (i < ColorTableDivider)
{
	_CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
}
else
{
	_CurrentBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
}
```

### My code idea
_See the complete code at the beginning of this comment!

We iterate the loop based on the number of colors of each type (8 non-bright and 8 bright colors) and use the variable "ColorTableDivider" as condition. This leads to 8 iterations with an index form 0 - 7. Then we don't get a higher index as we have elements in each of the "current color" lists.
```cpp
for (uint8_t i = 0; i < ColorTableDivider; ++i)
```

Within the loop we can use the loop's index to get the elements of the "current color" lists.
```cpp
_CurrentNonBrightColorTable.GetAt(i).Color(


_CurrentBrightColorTable.GetAt(i).Color(
```
Additionally, we know that we have 8 non-bright colors and 8 bright colors in the color scheme object's table. And we know that the bright color's element index is +8 elements higher as the related non-bright color's element index. So we can calculate the index for the bright color element based on the loop's index and the value of "ColorTableDivider". For the non-bright color element we can simply use the loop's index.
_Example:_
- _Index=0; non-bright color element=0; bright color element (index + ColorTableDivider)=8_
- _Index=7; non-bright color element=7; bright color element (index + ColorTableDivider)=15_
```cpp
int indexNonBrightColor = i;
    .Color(colorScheme.Table()[indexNonBrightColor]);
			
int indexBrightColor = i + ColorTableDivider;
	.Color(colorScheme.Table()[indexBrightColor]);
```

Reply:
@htcfreek, great, thank you for your thorough explanation! I’ll take a look soon.